internal/util/function/bedrock_embedding_provider.go

```diff
@@ -0,0 +1,209 @@
+/*
+ * # Licensed to the LF AI & Data foundation under one
+ * # or more contributor license agreements. See the NOTICE file
+ * # distributed with this work for additional information
+ * # regarding copyright ownership. The ASF licenses this file
+ * # to you under the Apache License, Version 2.0 (the
+ * # "License"); you may not use this file except in compliance
+ * # with the License. You may obtain a copy of the License at
+ * #
+ * #     http://www.apache.org/licenses/LICENSE-2.0
+ * #
+ * # Unless required by applicable law or agreed to in writing, software
+ * # distributed under the License is distributed on an "AS IS" BASIS,
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/bedrockruntime"
+)
+
+type BedrockClient interface {
+	InvokeModel(ctx context.Context, params *bedrockruntime.InvokeModelInput, optFns ...func(*bedrockruntime.Options)) (*bedrockruntime.InvokeModelOutput, error)
+}
+
+type BedrockEmbeddingProvider struct {
+	fieldDim int64
+
+	client        BedrockClient
+	modelName     string
+	embedDimParam int64
+	normalize     bool
+
+	maxBatch   int
+	timeoutSec int
+}
+
+func createBedRockEmbeddingClient(awsAccessKeyId string, awsSecretAccessKey string, region string) (*bedrockruntime.Client, error) {
+	if awsAccessKeyId == "" {
+		awsAccessKeyId = os.Getenv(bedrockAccessKeyId)
+	}
+	if awsAccessKeyId == "" {
+		return nil, fmt.Errorf("Missing credentials. Please pass `aws_access_key_id`, or configure the %s environment variable in the Milvus service.", bedrockAccessKeyId)
+	}
+
+	if awsSecretAccessKey == "" {
+		awsSecretAccessKey = os.Getenv(bedrockSecretAccessKey)
+	}
+	if awsSecretAccessKey == "" {
+		return nil, fmt.Errorf("Missing credentials. Please pass `aws_secret_access_key`, or configure the %s environment variable in the Milvus service.", bedrockSecretAccessKey)
+	}
+	if region == "" {
+		return nil, fmt.Errorf("Missing region. Please pass `region` param.")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
+			awsAccessKeyId, awsSecretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return bedrockruntime.NewFromConfig(cfg), nil
+}
+
+func NewBedrockEmbeddingProvider(fieldSchema *schemapb.FieldSchema, functionSchema *schemapb.FunctionSchema, c BedrockClient) (*BedrockEmbeddingProvider, error) {
```

Comment:
param `c BedrockClient` is never provided values other nil? in which case do we expect the caller to pass in a client?